### PR TITLE
157 implement chat

### PR DESCRIPTION
**Pull request by Thibrac**
Chat complete with : 
- General chat
- Private message
- Notifications
- Block button
- Link in message 
- Username clickable a link to the profile

### CODE REVIEW

File: django/src/pages/templates/extends/base.html [hunk 1]
```diff
@@ -124,8 +146,239 @@
 		});
 	});
 
-	document.getElementById("logout-button-navbar")?.addEventListener("click", function (e) {
+	document.getElementById("logout-button-navbar").addEventListener("click", function (e) {
```
Comment (Neffi42) [inline django/src/pages/templates/extends/base.html:149]: why ?

File: django/src/chat/admin.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (aurlic) [inline django/src/chat/admin.py:3]: remove comment

**Review by Neffi42 — APPROVED** [review]

File: django/src/chat/models.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.db import models
+
+# Create your models here.
```
Comment (aurlic) [inline django/src/chat/models.py:3]: remove comment

File: django/src/chat/admin.py (hunk 2 above)
Comment (Neffi42) [reply django/src/chat/admin.py:3]: auto-generated by djagno, it's ok to leave it

File: nginx/src/css/chat.css [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+#chat-log {
+	flex-grow: 1;
+	padding: 10px;
+	border: none;
+	overflow-y: auto;
+	display: flex;
+	flex-direction: column;
+	height: calc(100% - 100px);
+	background-image: url('../assets/chat-background2.webp');
```
Comment (aurlic) [inline nginx/src/css/chat.css:41]: maybe add fallback background-color in case image doesn't load

File: nginx/src/css/chat.css [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+#chat-button {
+	background-color: #f3e2f0;
+	position: fixed;
+	bottom: 0;
+	right: 0;
+	z-index: 1000;
+	cursor: pointer;
+	width: 60px;
+	height: 60px;
+	margin-left: 60px/5;
```
Comment (aurlic) [inline nginx/src/css/chat.css:77]:
```suggestion
	margin-left: 12px
```

File: nginx/src/css/chat.css [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+	margin-right: 60px/5;
```
Comment (aurlic) [inline nginx/src/css/chat.css:78]:
```suggestion
	margin-left: 12px
```

File: nginx/src/css/chat.css (hunk 5 above)
Comment (aurlic) [reply nginx/src/css/chat.css:77]:
```suggestion
	margin-left: (sqrt(144) * 3 / 2) + ((100 - 4^2) / (36 / 6)) + ((6 * (3 + 1)) / sqrt(16))px;
```

**Review by Neffi42 — CHANGES_REQUESTED** [review]
Merge main

**Review by Neffi42 — APPROVED** [review]

**Review by aurlic — APPROVED** [review]
LGTM

File: nginx/src/css/chat.css (hunk 4 above)
Comment (Thibrac) [reply nginx/src/css/chat.css:41]: if the image does not load, the background will be the base grey, so it will be the same as the rest of the image

**Comment by croueAlec** [comment]
> ## [Quality Gate Passed](https://sonarcloud.io/dashboard?id=TheSecretOrganization_ft_transcendence&pullRequest=235) **Quality Gate passed**
> 
> Issues  [2 New issues](https://sonarcloud.io/project/issues?id=TheSecretOrganization_ft_transcendence&pullRequest=235&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)  [0 Accepted issues](https://sonarcloud.io/project/issues?id=TheSecretOrganization_ft_transcendence&pullRequest=235&issueStatuses=ACCEPTED)
> 
> Measures  [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=TheSecretOrganization_ft_transcendence&pullRequest=235&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true)  [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=TheSecretOrganization_ft_transcendence&pullRequest=235&metric=new_coverage&view=list)  [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=TheSecretOrganization_ft_transcendence&pullRequest=235&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=TheSecretOrganization_ft_transcendence&pullRequest=235)

have the 2 new issues been fixed ?

**Comment by croueAlec** [comment]
c'est bon pour moi

**Pull request merged**